### PR TITLE
feat(search-input): add clear button at search

### PR DESCRIPTION
**Pull request by nikosanif**
#### Changelog:
- Add clear button as input suffix
- Add tests for clear button

### CODE REVIEW

**Comment by nikosanif** [comment]
Hi @bampakoa ! Awesome work!

I made a pull request adding a clear button at main search input. I'll be happy if you can review and merge it. Feel free to refactor the code and make any required changes.

**Comment by bampakoa** [comment]
@nikosanif thanks very much for your PR 🙏 I would be more than happy to review it.

Could you please first make sure that your editor has not done any formatting changes that are not related to the feature? Thanks!

**Comment by nikosanif** [comment]
Thanks for the advice @bampakoa! I have reverted all formatting changes. I was using `Prettier` as code formatter. Do you use any specific formatter or you use the defaults from `.editorconfig`?

**Review by bampakoa — COMMENTED** [review]
@nikosanif I think that the tests for the clear button are not necessary since we are not interacting with the component class. Let me elaborate a bit on this:

The `searchBox` variable is a [template reference variable ](https://angular.io/guide/template-reference-variables) that is present only in the template of the component. It does not interact with a property in the TypeScript class of the component. Thus, these types of tests seem to be a better fit for e2e tests 😉

**Comment by bampakoa** [comment]
I am a fan of the default configuration generally 😁 so, yes, I use the ones from the `.editorconfig` (for now).

**Comment by nikosanif** [comment]
> @nikosanif I think that the tests for the clear button are not necessary since we are not interacting with the component class. Let me elaborate a bit on this:
> 
> The `searchBox` variable is a [template reference variable ](https://angular.io/guide/template-reference-variables) that is present only in the template of the component. It does not interact with a property in the TypeScript class of the component. Thus, these types of tests seem to be a better fit for e2e tests 😉 

Another good advice for me 😁! I don't have much experience about tests so your statement it's very useful. 

Do think that it's better to remove them completely?

File: src/app/characters/character-list/character-list.component.html [hunk 1]
```diff
@@ -14,6 +14,16 @@
...
+        <button
+          *ngIf="searchBox?.value"
+          mat-button
+          matSuffix
+          mat-icon-button
+          aria-label="Clear"
+          (click)="searchBox.value = ''"
+        >
+          <mat-icon>close</mat-icon>
+        </button>
```
[the remark this answers is not in the document]
Comment (nikosanif) [reply src/app/characters/character-list/character-list.component.html]: Yep, I changed it to single line.

**Comment by bampakoa** [comment]
>Do think that it's better to remove them completely?

If you can, that would be great. We can add it back l8r when starting to write e2e tests 😃

File: src/app/characters/character-list/character-list.component.html [hunk 2]
```diff
@@ -14,6 +14,9 @@
     <div class="search">
       <mat-form-field>
         <input matInput placeholder="Search by name" #searchBox (keyup)="search(searchBox.value)">
+        <button *ngIf="searchBox?.value" mat-button matSuffix mat-icon-button aria-label="Clear" (click)="searchBox.value = ''">
```
Comment (bampakoa) [inline src/app/characters/character-list/character-list.component.html:17]:
```suggestion
        <button *ngIf="searchBox?.value" matSuffix mat-icon-button aria-label="Clear" (click)="searchBox.value = ''">
```

**Review by bampakoa — CHANGES_REQUESTED** [review]
@nikosanif just a minor change and we are good to go 😃

File: src/app/characters/character-list/character-list.component.html (hunk 2 above)
Comment (nikosanif) [reply src/app/characters/character-list/character-list.component.html]: Nice catch! I didn't see that.

**Review by bampakoa — APPROVED** [review]

**Pull request merged**